### PR TITLE
[crowdstrike.fdr] Add error handling for rename processor in pipeline

### PR DESCRIPTION
**Pull request by monkeyonbranch**
Fixes error handling for renaming processor for user.name field.

## Proposed commit message
Error handling for rename processor regarding user. name field.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

### CODE REVIEW

**Review by efd6 — COMMENTED** [review]
This needs changes to the changelog and package manifest.

Is there an issue for this? Also it would be good if you could add a test case that displays the issue.

**Comment by monkeyonbranch** [comment]
The issue is in the beats repository but it affects this pipeline: https://github.com/elastic/beats/issues/39360

Screenshot 2024-05-06 at 09 21 59

File: packages/crowdstrike/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.33.1"
+  changes:
+    - description: Improve error handling for renaming processors.
+      type: enhancement
```
Comment (efd6) [inline packages/crowdstrike/changelog.yml:5]:
```suggestion
      type: bugfix
```

**Review by efd6 — COMMENTED** [review]
We need the manifest change as well and a test case. The issue doesn't include a repro (thanks for linking, I have transferred it to this repo).

**Comment by monkeyonbranch** [comment]
I'm quite new to this so I'm a bit lost as how to do this.

**Comment by efd6** [comment]
OK. The tool we use is [elastic-package](https://github.com/elastic/elastic-package/). If you install that you will be able to run tests and so on. You will need to collect an event that would fail without this change. This is best done by turning on "Preserve Original Event" in the UI and then collecting the string from the field. This can then be unquoted and added into the tests in [this file](https://github.com/elastic/integrations/blob/main/packages/crowdstrike/data_stream/fdr/_dev/test/pipeline/test-fdr.log). Then run `elastic-package test pipeline -g` to regenerate the expectations.

Unfortunately it difficult for me to do this for you because you are attempting to merge in from you main branch. In future, it's best to create a new branch to work on in order to send changes.

**Comment by monkeyonbranch** [comment]
First off, thank you for the guidance. I got it to work! Very easy to use!
So, using the crowdstrike json object in the document that throws an error in our deployment doesn't throw an error when testing it locally. 

The pipeline has the follwing two processors:
`
  {
    "rename": {
      "field": "crowdstrike.info.user.UserName",
      "target_field": "user.name",
      "ignore_missing": true
    }
  },`
and
`
  {
    "rename": {
      "field": "crowdstrike.UserName",
      "target_field": "user.name",
      "ignore_missing": true
    }
  },`

As you can see they reference the fields crowdstrike.UserName and crowdstrike.info.user.UserName. Well, in our documents the second field doesn't exist. The field that does exist is called crowdstrike.info.user.User. 
When I rename crowdstrike.info.user.User into crowdstrike.info.user.UserName in the test-fdr.log the elastic-package test throws the same error we get in our deployment. 

I can't really get my head around that. It seems we're getting an error thrown because of a field that doesn't exist.

**Comment by monkeyonbranch** [comment]
I created my own branch and commited my changes to it. I will try and understand how to replace the code in this PR with the new one:
https://github.com/monkeyonbranch/integrations/tree/fix-rename-processors-crowdstrike_fdr

**Comment by efd6** [comment]
You'll need to make the following changes to you other branch.
```diff
diff --git a/packages/crowdstrike/data_stream/fdr/_dev/test/pipeline/test-fdr.log-expected.json b/packages/crowdstrike/data_stream/fdr/_dev/test/pipeline/test-fdr.log-expected.json
index 8c756bf84..3d110d9e9 100644
--- a/packages/crowdstrike/data_stream/fdr/_dev/test/pipeline/test-fdr.log-expected.json
+++ b/packages/crowdstrike/data_stream/fdr/_dev/test/pipeline/test-fdr.log-expected.json
@@ -11867,6 +11867,101 @@
             "user": {
                 "name": "user9"
             }
+        },
+        {
+            "@timestamp": "2024-05-07T10:46:39.690Z",
+            "crowdstrike": {
+                "AuthenticationId": "111112312312312321",
+                "ConfigStateHash": "821711964",
+                "EffectiveTransmissionClass": "2",
+                "Entitlements": "15",
+                "LogonTime": "2024-05-07T10:46:39.631Z",
+                "LogonType": "2",
+                "PasswordLastSet": "1706789855.733",
+                "UserGroupsBitmask": "0",
+                "UserLogonFlags": "6",
+                "UserName": "user-1",
+                "cid": "1d1d1d1d1d1d1d1d1",
+                "info": {
+                    "host": {
+                        "AgentLoadFlags": "0",
+                        "AgentLocalTime": "1714620029.775",
+                        "AgentTimeOffset": "454032",
+                        "AgentVersion": "7.14.18305.0",
+                        "ChassisType": "Other",
+                        "City": "Bengaluru",
+                        "ConfigBuild": "1007.4.0018305.10",
+                        "ConfigIDBuild": "18305",
+                        "Continent": "Asia",
+                        "Country": "India",
+                        "FirstSeen": "1701672920.000",
+                        "HostHiddenStatus": "Visible",
+                        "SystemManufacturer": "Apple Inc.",
+                        "SystemProductName": "Mac14, 2",
+                        "Time": "1715074061.000",
+                        "Timezone": "Asia/Kolkata",
+                        "Version": "Ventura (13)",
+                        "cid": "11d1d1d1dd1d1d1d1d1",
+                        "event_platform": "Mac"
+                    },
+                    "user": {
+                        "AccountType": "Local",
+                        "LastLoggedOnHost": "User-1-MacBook-Air.local",
+                        "LogonTime": "1715076000.000",
+                        "LogonType": "Interactive",
+                        "PasswordLastSet": "1706789855.733",
+                        "UserIsAdmin": "1",
+                        "_time": "1715076916.290",
+                        "cid": "1d1d1d1d1d1d1d1d1",
+                        "monthsincereset": "3.0"
+                    }
+                },
+                "name": "UserLogonMacV1"
+            },
+            "ecs": {
+                "version": "8.11.0"
+            },
+            "event": {
+                "created": "2024-05-07T10:46:39.943Z",
+                "original": "{\"UserIsAdmin\": \"1\", \"LogonTime\": \"1715078799.631\", \"UserSid\": \"S-1-5-21-1111-11111-387821029-2004\", \"event_platform\": \"Mac\", \"EffectiveTransmissionClass\": \"2\", \"timestamp\": \"1715078799943\", \"info\": { \"host\": { \"Timezone\": \"Asia/Kolkata\", \"ServicePackMajor\": \"none\", \"PointerSize\": \"none\", \"SiteName\": \"none\", \"AgentVersion\": \"7.14.18305.0\", \"Time\": \"1715074061.000\", \"MachineDomain\": \"none\", \"event_platform\": \"Mac\", \"AgentTimeOffset\": \"454032\", \"Version\": \"Ventura (13)\", \"AgentLoadFlags\": \"0\", \"BiosVersion\": \"none\", \"FalconGroupingTags\": \"none\", \"SystemProductName\": \"Mac14, 2\", \"OU\": \"none\", \"ProductType\": \"none\", \"City\": \"Bengaluru\", \"ConfigBuild\": \"1007.4.0018305.10\", \"SensorGroupingTags\": \"none\", \"Continent\": \"Asia\", \"FirstSeen\": \"1701672920.000\", \"HostHiddenStatus\": \"Visible\", \"BiosManufacturer\": \"none\", \"AgentLocalTime\": \"1714620029.775\", \"Country\": \"India\", \"ChassisType\": \"Other\", \"SystemManufacturer\": \"Apple Inc.\", \"ConfigIDBuild\": \"18305\", \"cid\": \"11d1d1d1dd1d1d1d1d1\" }, \"user\": { \"UserIsAdmin\": \"1\", \"monthsincereset\": \"3.0\", \"User\": \"USER-1-MACBOOK-AIR.LOCAL\\\\USER-1\", \"LogonTime\": \"1715076000.000\", \"PasswordLastSet\": \"1706789855.733\", \"LogonType\": \"Interactive\", \"AccountType\": \"Local\", \"LastLoggedOnHost\": \"User-1-MacBook-Air.local\", \"_time\": \"1715076916.290\", \"cid\": \"1d1d1d1d1d1d1d1d1\" } }, \"ContextTimeStamp\": \"1715078799.690\", \"UserName\": \"user-1\", \"ConfigStateHash\": \"821711964\", \"LogonType\": \"2\", \"AuthenticationId\": \"111112312312312321\", \"UserPrincipal\": \"user-1@User-1-MacBook-Air.local\", \"PasswordLastSet\": \"1706789855.733\", \"Entitlements\": \"15\", \"name\": \"UserLogonMacV1\", \"UserLogonFlags\": \"6\", \"UserGroupsBitmask\": \"0\", \"cid\": \"1d1d1d1d1d1d1d1d1\"}"
+            },
+            "host": {
+                "os": {
+                    "type": "macos"
+                }
+            },
+            "observer": {
+                "type": "agent",
+                "vendor": "crowdstrike"
+            },
+            "related": {
+                "hash": [
+                    "821711964"
+                ],
+                "hosts": [
+                    "User-1-MacBook-Air.local"
+                ],
+                "user": [
+                    "USER-1-MACBOOK-AIR.LOCAL\\USER-1",
+                    "user-1"
+                ]
+            },
+            "tags": [
+                "preserve_original_event"
+            ],
+            "url": {
+                "scheme": "http"
+            },
+            "user": {
+                "domain": "User-1-MacBook-Air.local",
+                "email": "user-1@User-1-MacBook-Air.local",
+                "full_name": "user-1",
+                "id": "S-1-5-21-1111-11111-387821029-2004",
+                "name": "USER-1-MACBOOK-AIR.LOCAL\\USER-1",
+                "roles": [
+                    "admin"
+                ]
+            }
         }
     ]
 }
\ No newline at end of file
diff --git a/packages/crowdstrike/data_stream/fdr/fields/fields.yml b/packages/crowdstrike/data_stream/fdr/fields/fields.yml
index 434862854..5f67ea034 100644
--- a/packages/crowdstrike/data_stream/fdr/fields/fields.yml
+++ b/packages/crowdstrike/data_stream/fdr/fields/fields.yml
@@ -448,6 +448,8 @@
       type: long
     - name: UserMemoryProtectExecutableRemoteCount
       type: long
+    - name: UserName
+      type: keyword
     - name: UserTime
       type: long
     - name: VnodeModificationType
```
It's fine to close this and open a new PR with the other branch if you want. Otherwise, just doing a `git reset --hard fix-rename-processors-crowdstrike_fdr` when you are on your main branch, and then applying the diff I showed above will work.

**Comment by monkeyonbranch** [comment]
I created a new PR from my new branch. Figured it'd make things a bit more tidy.
https://github.com/elastic/integrations/pull/9816

**Pull request closed without merging**